pkg/resource/role/hooks.go

```diff
@@ -382,6 +394,29 @@ func compareTags(
 	}
 }
 
+// compareAssumeRolePolicyDocument is a custom comparison function for
+// assumeRolePolicyDocuments. The reason why we need a custom function for
+// this fields is the API logic that trims all the trailing while spaces
+// string of provided documents.
+func compareAssumeRolePolicyDocument(
```

Comment:
TIL, that not only the IAM API trims the spaces and new lines, but it also transforms `[]string` to `string` if they contain one element........
/hold